discovery-provider/tests/test_index_playlists.py

```diff
@@ -0,0 +1,301 @@
+            entry,
+            event_type,
+            playlist_record,
+            12, # block_timestamp,
+            session
+        )
+
+        assert len(playlist_record.playlist_contents['track_ids']) == 1
+        last_playlist_content = playlist_record.playlist_contents['track_ids'][-1]
+        assert last_playlist_content == {"track": entry.args._addedTrackId, "time": 12}
+
+        # ================= Test playlist_track_added with second track Event =================
+        event_type, entry = get_playlist_track_added_event(1, 2)
+
+        parse_playlist_event(
+            None, # self - not used
+            None, # update_task - not used
+            entry,
+            event_type,
+            playlist_record,
+            13, # block_timestamp,
+            session
+        )
+
+        assert len(playlist_record.playlist_contents['track_ids']) == 2
+        last_playlist_content = playlist_record.playlist_contents['track_ids'][-1]
+        assert last_playlist_content == {"track": entry.args._addedTrackId, "time": 13}
+
+        # ================= Test playlist_tracks_ordered Event =================
+        event_type, entry = get_playlist_tracks_ordered_event()
+        parse_playlist_event(
+            None, # self - not used
+            None, # update_task - not used
+            entry,
+            event_type,
+            playlist_record,
+            block_timestamp,
+            session
+        )
+
+        assert playlist_record.playlist_contents['track_ids'] == [
+            {"track": 2, "time": 13},
+            {"track": 1, "time": 12}
+        ]
+
+        # ================= Test playlist_track_delete_event Event =================
+        event_type, entry = get_playlist_track_delete_event(1, 1, 12)
+
+        parse_playlist_event(
+            None, # self - not used
+            None, # update_task - not used
+            entry,
+            event_type,
+            playlist_record,
+            block_timestamp,
+            session
+        )
+
+        assert len(playlist_record.playlist_contents['track_ids']) == 1
+        last_playlist_content = playlist_record.playlist_contents['track_ids'][-1]
+        assert playlist_record.playlist_contents['track_ids'] == [{"track": 2, "time": 13}]
+
+        # ================= Test playlist_track_delete_event Event =================
+        # This should be a no-op
+        event_type, entry = get_playlist_track_delete_event(1, 1, 12)
+
+        parse_playlist_event(
+            None, # self - not used
+            None, # update_task - not used
+            entry,
+            event_type,
+            playlist_record,
+            block_timestamp,
+            session
+        )
+
+        assert len(playlist_record.playlist_contents['track_ids']) == 1
+        assert playlist_record.playlist_contents['track_ids'] == [{"track": 2, "time": 13}]
+
+        # ================= Test playlist_deleted Event =================
```

Comment:
feels like these should be separate tests entirely with a helper/fixture that does all of the setup.
ok w/ this as is, but it might make life easier to figure out what's broken if this test fails